atlasdb-config/src/main/java/com/palantir/atlasdb/http/TextDelegateDecoder.java

```diff
@@ -59,15 +57,13 @@ public TextDelegateDecoder(Decoder delegate) {
     }
 
     @Override
-    public Object decode(Response response, Type type) throws IOException, DecodeException, FeignException {
-        Collection<String> contentTypes = response.headers().get(HttpHeaders.CONTENT_TYPE);
-        // In the case of multiple content types, or an unknown content type, we'll use the delegate instead.
-        if (contentTypes != null
-                && contentTypes.size() == 1
-                && Iterables.getOnlyElement(contentTypes, "").equals(MediaType.TEXT_PLAIN)) {
+    public Object decode(Response response, Type type) throws IOException, FeignException {
+        if (HeaderAccessUtils.shortcircuitingCaseInsensitiveContainsEntry(
+                response.headers(),
+                HttpHeaders.CONTENT_TYPE,
```

Comment:
I'm wondering if it is worth using a pre-upper case header, as the current `equalsIgnoreCase` implementation converts chars to upper case for comparison, though this might be an over-optimization to the current `equalsIgnoreCase` implementation, but JMH benchmark would help show whether its worth it:

```
    private static final String CONTENT_TYPE = HttpHeaders.CONTENT_TYPE.toUpperCase();
...
        if (HeaderAccessUtils.shortcircuitingCaseInsensitiveContainsEntry(
                response.headers(),
                CONTENT_TYPE,
                MediaType.TEXT_PLAIN)) {
            return stringDecoder.decode(response, type);
        }
```